examples/plot_text_classification.py

```diff
@@ -0,0 +1,169 @@
+
+hub_utils.init(
+    model=pkl_name,
+    requirements=[f"scikit-learn={sklearn.__version__}"],
+    dst=local_repo,
+    task="text-classification",
+    data=X_test,
+)
+
+# %% Create a model card
+# ====================
+# We now create a model card, and populate its metadata with information which
+# is already provided in ``config.json``, which itself is created by the call to
+# :func:`.hub_utils.init` above. We will see below how we can populate the model
+# card with useful information.
+
+model_card = card.Card(model, metadata=card.metadata_from_config(Path(local_repo)))
+
+# %% Add more information
+# ====================
+# So far, the model card does not tell viewers a lot about the model. Therefore,
+# we add more information about the model, like a description and what its
+# license is.
+
+model_card.metadata.license = "mit"
+limitations = "This model is not ready to be used in production."
+model_description = (
+    "This is a Multinomial Naive Bayes model trained on 20 news groups dataset."
+    "Count vectorizer and TFIDF vectorizer are used for vectorization."
+)
+model_card_authors = "skops_user"
+get_started_code = (
+    "import pickle\nwith open(pkl_filename, 'rb') as file:\n    clf = pickle.load(file)"
+)
+citation_bibtex = "bibtex\n@inproceedings{...,year={2020}}"
+model_card.add(
+    citation_bibtex=citation_bibtex,
+    get_started_code=get_started_code,
+    model_card_authors=model_card_authors,
+    limitations=limitations,
+    model_description=model_description,
+)
+
+# %% Add plots, metrics, and tables to our model card
+# ================================================
+# We will evaluate our model
+
+y_pred = model.predict(X_test)
+eval_descr = (
+    "The model is evaluated on validation data from 20 news group's test split,"
+    " using accuracy and F1-score with micro average."
+)
+model_card.add(eval_method=eval_descr)
+
+accuracy = accuracy_score(y_test, y_pred)
+f1 = f1_score(y_test, y_pred, average="micro")
+model_card.add_metrics(**{"accuracy": accuracy, "f1 score": f1})
+
+cm = confusion_matrix(y_test, y_pred, labels=model.classes_)
+disp = ConfusionMatrixDisplay(confusion_matrix=cm, display_labels=model.classes_)
+disp.plot()
+
+disp.figure_.savefig(Path(local_repo) / "confusion_matrix.png")
+model_card.add_plot(**{"Confusion matrix": "confusion_matrix.png"})
+
+clf_report = classification_report(
+    y_test, y_pred, output_dict=True, target_names=categories
+)
+# The classification report has to be transformed into a DataFrame first to have
+# the correct format. This requires removing the "accuracy", which was added
+# above anyway.
+del clf_report["accuracy"]
+clf_report = pd.DataFrame(clf_report).T.reset_index()
+model_card.add_table(
+    folded=True,
+    **{
+        "Classification report": clf_report,
+    },
+)
+
+# %% Save model card
+# ===============
+# We can simply save our model card by providing a path to :meth:`.Card.save`.
+
+model_card.save(Path(local_repo) / "README.md")
```

Comment:
we don't have to add the code to this example, but it might be useful to add a comment that this so far hasn't added the model to the hub, and if the user wants to add them to the hub, they should refer to the other example, or call push. WDYT?

Reply:
and here